distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso

```diff
@@ -91,11 +93,16 @@ read : Text | URI | File -> File_Format -> Problem_Behavior -> Any ! File_Error
 read path=(Missing_Argument.throw "path") format=Auto_Detect (on_problems : Problem_Behavior = ..Report_Warning) = case path of
     _ : Text -> if Data_Read_Helpers.looks_like_uri path then Data_Read_Helpers.fetch_following_data_links path format=format else
         read (File.new path) format on_problems
-    uri : URI -> fetch uri format=format
+    uri : URI ->
+        response = fetch uri format=format
+        Telemetry.log "Data.read" "fetching from URI" (Dictionary.from_vector [["content_length", response.content_length]])
+        response
     _ ->
         file_obj = File.new path
         if file_obj.is_directory then Error.throw (Illegal_Argument.Error "Cannot `read` a directory, use `Data.list`.") else
-            file_obj.read format on_problems
+            file_content = file_obj.read format on_problems
+            Telemetry.log "Data.read" "read from file" (Dictionary.from_vector [["size", file_obj.size]])
```

Comment:
@jdunkerley mentioned that the content needs to be "extensible". There are two options:
- either you log basic info here and let `file_obj.read` implementation also log something to the telemetry
- or you obtain a `Dictionary` from the `file_obj.read` operation, merge it with keys obtained here and log all at once

Reply:
> either you log basic info here and let file_obj.read implementation also log something to the telemetry

This is the simplest approach at the moment, but cannot be implemented as long as `Telemetry` is a private module inside `Standard.Base`.

> or you obtain a Dictionary from the file_obj.read operation, merge it with keys obtained here and log all at once

This feels unnecessarily complicated. And will get outdated the minute we implement telemetry collection via appender. Let's keep the telemetry collection in this PR as simple as possible, and let's just keep logging the byte size of the data read. I will try to also include at least the MIME type, or the extension of the file.

Reply:
- Yes, include mime type.
- Possibly include also Enso type of the read object
- Right, until there is appender and/or while the `Telemetry` remains `private` things are restricted
- however that's perfectly OK if things **remain restricted** in this PR
- we'll address the extensibility in subsequent PRs